components/Form/Register.tsx

```diff
@@ -0,0 +1,128 @@
+import AsyncStorage from "@react-native-async-storage/async-storage";
+import React, { useState } from "react";
+import { View, Text, TextInput, Pressable, Alert } from "react-native";
+
+interface FormProps {
+  onNavigate: () => void;
+}
+
+export default function Register({ onNavigate }: FormProps) {
+  const [name, setName] = useState("");
+  const [password, setPassword] = useState("");
+  const [email, setEmail] = useState("");
+
+  const checkExistingUser = async (userEmail: string) => {
+    try {
+      const existingUsers = await AsyncStorage.getItem("registeredUsers");
+      if (existingUsers) {
+        const users = JSON.parse(existingUsers);
+        return users.some(
+          (user: { email: string }) => user.email === userEmail
+        );
+      }
+      return false;
+    } catch (error) {
+      console.error("Failed to fetch the users", error);
+      return false;
+    }
+  };
+
+  const saveRegisterData = async () => {
+    try {
+      const userExists = await checkExistingUser(email);
+
+      if (userExists) {
+        Alert.alert("Error", "User with this email already exists");
+        return false;
+      }
+
+      const existingUsers = await AsyncStorage.getItem("registeredUsers");
+      const users = existingUsers ? JSON.parse(existingUsers) : [];
+      users.push({ name, email, password });
+
+      await AsyncStorage.setItem("registeredUsers", JSON.stringify(users));
+      return true;
+    } catch (error) {
+      console.error("Failed to fetch Data", error);
+    }
+  };
+
+  const handleSubmit = async () => {
+    if (!name || !email || !password) {
+      Alert.alert("Pleaase fill all the required filled");
+      return;
+    }
+
+    if (password.length < 8) {
```

Comment:
Minimal input validation. Only checks for empty fields.

Add robust validation for email format and password strength using regex or a library like [validator.js](https://github.com/validatorjs/validator.js).